api/server/response_options.proto

```diff
@@ -5,6 +5,15 @@ package nighthawk.server;
 import "google/protobuf/wrappers.proto";
 import "validate/validate.proto";
 import "envoy/api/v2/core/base.proto";
+import "google/protobuf/duration.proto";
+
+message GaugeBasedDelay {
```

Comment:
I feel like this name focuses on the mechanism of how it's implemented, and it would be more intuitive to instead focus on the value it provides. Something more like: `QueuingDelay` if we want to be very concise, or `DelayComputedFromQueueSize` for a more verbose version.

On the other hand I see you are generically allowing computing a delay from any gauge, and I'm curious if you've thought of a use-case other than queued requests, for this generality to be helpful?